ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-bgp.j2

```diff
@@ -44,20 +44,51 @@
 | {{ paths_cli }} |
 {%         endif %}
 {%     endif %}
-{%     if router_bgp.listen_ranges is arista.avd.defined %}
+{# Check if neighbors or listen_ranges exist under router_bgp.vrfs #}
+{%     set temp = namespace() %}
+{%     set temp.bgp_vrf_neighbors = false %}
+{%     set temp.bgp_vrf_listen_ranges = false %}
+{%     if router_bgp.vrfs is arista.avd.defined %}
+{%         for vrf in router_bgp.vrfs | arista.avd.convert_dicts('name') | arista.avd.natural_sort('name') %}
+{%             if vrf.neighbors is arista.avd.defined %}
+{%                 set temp.bgp_vrf_neighbors = true %}
```

Comment:
After internal discussion, though this loop generating both `temp.bgp_vrf_neighbors` and `temp.bgp_vrf_listen_ranges` is more efficient from a computation point of view, the current design decision  is to keep the loop generating the temp variable above the section where it is used for ease of comprehension and maintainability sake. 

Could you please keep the loop you removed https://github.com/aristanetworks/ansible-avd/pull/1779/files#diff-95771e6e994ce349aaadc66cbc36c4536b3b43849628f702ec39b2b1cc5f0df5L162 and not generate the `temp.bgp_vrf_neighbors` loop there ?

Reply:
sure, let me change it.. depending on the "amount" of vrfs and/or neighbors it might even be faster to have it sep'ed (because this allows us to keep using a break to end the loop on first hit)..